diff --git a/node_modules/cheerio/lib/static.js b/node_modules/cheerio/lib/static.js
index 840f2ae..d09c623 100644
--- a/node_modules/cheerio/lib/static.js
+++ b/node_modules/cheerio/lib/static.js
@@ -18,7 +18,7 @@ var serialize = require('dom-serializer'),
 
 exports.load = function(content, options, isDocument) {
   var Cheerio = require('./cheerio');
-
+  defaultOptions.serialize = serialize
   options = _.defaults(flattenOptions(options || {}), defaultOptions);
 
   if (isDocument === void 0)
@@ -72,7 +72,7 @@ function render(that, dom, options) {
     dom = select(dom, that._root, options);
   }
 
-  return serialize(dom, options);
+  return options.serialize(dom, options);
 }
 
 /**
@@ -92,7 +92,7 @@ exports.html = function(dom, options) {
 
   // sometimes $.html() used without preloading html
   // so fallback non existing options to the default ones
-  options = _.defaults(flattenOptions(options || {}), this._options, defaultOptions);
+  options = _.defaults(flattenOptions(options || {}), this._options, defaultOptions, {xmlMode: true, spaceBeforeSelfClose: true});
 
   return render(this, dom, options);
 };
