diff --git a/node_modules/dom-serializer/index.js b/node_modules/dom-serializer/index.js
index 4f41e8f..2dd69de 100644
--- a/node_modules/dom-serializer/index.js
+++ b/node_modules/dom-serializer/index.js
@@ -33,8 +33,8 @@ function formatAttrs(attributes, opts) {
     }
 
     output += key;
-    if ((value !== null && value !== '') || opts.xmlMode) {
-        output += '="' + (opts.decodeEntities ? entities.encodeXML(value) : value) + '"';
+    if ((value !== null && value !== '') || (opts.xmlMode && !booleanAttributes[key])) {
+        output += '="' + value + '"';
     }
   }
 
@@ -66,6 +66,52 @@ var singleTag = {
   track: true,
   wbr: true,
 };
+var nestedTag = {
+  div: true,
+  span: true,
+  textarea: true,
+  script: true,
+};
+
+var booleanAttributes = {
+  async: true,
+  autocomplete: true,
+  autofocus: true,
+  autoplay: true,
+  border: true,
+  challenge: true,
+  checked: true,
+  compact: true,
+  contenteditable: true,
+  controls: true,
+  default: true,
+  defer: true,
+  disabled: true,
+  formNoValidate: true,
+  frameborder: true,
+  hidden: true,
+  indeterminate: true,
+  ismap: true,
+  loop: true,
+  multiple: true,
+  muted: true,
+  nohref: true,
+  noresize: true,
+  noshade: true,
+  novalidate: true,
+  nowrap: true,
+  open: true,
+  readonly: true,
+  required: true,
+  reversed: true,
+  scoped: true,
+  scrolling: true,
+  seamless: true,
+  selected: true,
+  sortable: true,
+  spellcheck: true,
+  translate: true,
+}
 
 
 var render = module.exports = function(dom, opts) {
@@ -96,7 +142,7 @@ var render = module.exports = function(dom, opts) {
 
 function renderTag(elem, opts) {
   // Handle SVG
-  if (elem.name === "svg") opts = {decodeEntities: opts.decodeEntities, xmlMode: true};
+  if (elem.name === "svg") opts = Object.assign(opts, {decodeEntities: opts.decodeEntities, xmlMode: true});
 
   var tag = '<' + elem.name,
       attribs = formatAttrs(elem.attribs, opts);
@@ -106,18 +152,27 @@ function renderTag(elem, opts) {
   }
 
   if (
-    opts.xmlMode
-    && (!elem.children || elem.children.length === 0)
+    singleTag[elem.name]
+    || (opts.xmlMode
+    && (!elem.children || elem.children.length === 0))
   ) {
-    tag += '/>';
+    if (nestedTag[elem.name]) {
+      tag += '></' + elem.name + '>';
+    } else {
+      if (opts.spaceBeforeSelfClose) tag += ' ';
+      tag += '/>';
+    }
+    if (/link|meta/.test(elem.name)) tag += '\n'
   } else {
     tag += '>';
+    if (/html|head|body/.test(elem.name)) tag += '\n'
     if (elem.children) {
       tag += render(elem.children, opts);
     }
-
-    if (!singleTag[elem.name] || opts.xmlMode) {
+    if (/body/.test(elem.name)) tag += '\n'
+    if (!singleTag[elem.name]) {
       tag += '</' + elem.name + '>';
+      if (/html|head|body/.test(elem.name)) tag += '\n'
     }
   }
 
